CMakeLists.txt

```diff
@@ -602,43 +597,26 @@ endif()
 # Unit tests
 ################################################################################
 if (BUILD_CPPUNIT_TEST)
+    find_package(Catch2 REQUIRED)
+
     set(UNIT_TESTS
-        Utest/testEffectiveDistanceToEllipse.cpp
-        Utest/testEffectiveDistanceToLine.cpp
-        Utest/testGetArea.cpp
-        Utest/testIsInside.cpp
-        Utest/testIsOn.cpp
-        Utest/testIsOutside.cpp
-        Utest/testPointOnEllipse.cpp
-        Utest/testSign.cpp
-        Utest/testSign1.cpp
-    )
-    foreach(file ${UNIT_TESTS})
-        get_filename_component(test ${file} NAME_WE)
-        add_executable(${test} ${file})
-        target_link_libraries(${test} core)
-        target_compile_options(${test} PRIVATE
-            ${COMMON_COMPILE_OPTIONS}
-        )
-        add_test(NAME ${test} COMMAND $<TARGET_FILE:${test}>)
-    endforeach()
-    add_executable(boost-unit-tests
-        Utest/test_geometry/testClassLineBoost.cpp
-        Utest/test_geometry/testClassObstacleBoost.cpp
-        Utest/test_geometry/testClassPointBoost.cpp
-        Utest/test_geometry/testClassRoomBoost.cpp
-        Utest/test_geometry/testClassWallBoost.cpp
-        Utest/test_geometry/testMain.cpp
-        Utest/test_geometry/testSubRoomBoost.cpp
+      test/catch2/Main.cpp
     )
-    target_link_libraries(boost-unit-tests
-        core
-        Boost::unit_test_framework
+
+    add_executable(unittests
+        ${UNIT_TESTS}
```

Comment:
You can simply add the files you want to compile here, no need to  set this as a variable and then expand the variable. This was necessary because there where many executables build but this should not be necessary anymore if you restructure with catch2.